### PR TITLE
add new references to learn component lifecycles

### PR DESCRIPTION
**Pull request by VANSH3104**
The previous references for component lifecycles were outdated as they focused on the class component paradigm. This PR updates the documentation to include a more relevant reference that explains the lifecycle methods in the context of functional components, aligning with modern React practices. https://github.com/kamranahmedse/developer-roadmap/issues/6771

### CODE REVIEW

**Comment by VANSH3104** [comment]
@dansholds, I have made a proper pull request. I apologize for the inconvenience caused by the previous one.

File: src/data/roadmaps/react/content/component-lifecycle@8OBlgDRUg-CTgDXY-QHyO.md [hunk 1]
```diff
@@ -6,5 +6,8 @@ It is not recommended to use lifecycle methods manually. Instead, use the useEff
 
 Visit the following resources to learn more:
 
-- [@official@Class Component](https://react.dev/reference/react/Component)
-- [@official@Lifecycle of Reactive Effects](https://react.dev/learn/lifecycle-of-reactive-effects)
+- [@article@React component lifecycle: React lifecycle methods & hooks](https://tsh.io/blog/react-component-lifecycle-methods-vs-hooks/)
+- [@article@The React lifecycle: methods and hooks explained](https://retool.com/blog/the-react-lifecycle-methods-and-hooks-explained#react-hooks-and-the-component-lifecycle)
+- [@article@React Lifecycle: Methods & Hooks In Detail](https://www.bairesdev.com/blog/react-lifecycle-methods-hooks/)
+- [@official@Lifecycle of Reactive Effects](https://react.dev/learn/lifecycle-of-reactive-effectsv)
+- [@official@Class Component](https://react.dev/reference/react/Component)
```
Comment (offensive-vk) [inline src/data/roadmaps/react/content/component-lifecycle@8OBlgDRUg-CTgDXY-QHyO.md:13]: Official links must have higher priority, then the others and 2 article links are sufficient.

File: src/data/roadmaps/react/content/component-lifecycle@8OBlgDRUg-CTgDXY-QHyO.md
Comment (offensive-vk) [inline src/data/roadmaps/react/content/component-lifecycle@8OBlgDRUg-CTgDXY-QHyO.md:1]: There should be no links in main para, just keep the links below if required.
In the mean time, have a thorough understanding of our [contributing guidelines](https://github.com/kamranahmedse/developer-roadmap/blob/master/contributing.md).

**Comment by offensive-vk** [comment]
Lastly, please refrain from pinging maintainers, all PR will be actioned.

**Review by dansholds — CHANGES_REQUESTED** [review]
I know you didn't add but please remove the main body link.

**Comment by VANSH3104** [comment]
i gave it a article tag because other tags don't hold on it

**Review by dansholds — APPROVED** [review]

**Pull request merged**